docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,216 @@
+
+## Motivation
+
+Data provides insight in order to improve the product for our users.
+
+### Goals
+
+1. Create a process to send [event](https://github.com/openshift/assisted-service/tree/master/docs/dev/events.md) data from on-premises deployments of AI.
+2. Analyze and process on-premises data (by use of visualizations, dashboards, etc.) at Red Hat similarly to SaaS data.
+3. Analyze on-premises data both separately and together with SaaS data.
+
+
+### Non-Goals
+
+1. Implementing new data types to collect is out of the scope of this epic. This epic defines how and when the data already collected will be processed and sent.
+2. Define how SaaS should redesign their data pipeline (taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)) is out of the scope of this epic. This epic only refers to on-premises deployments of AI.
+3. Define the creation and/or management of an event streaming instance such as Kafka and the endpoint this data will be hosted at (both taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)).
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+5. Data formatting and handling by the event streaming service is out of the scope of this epic. The event streaming service should be able to process the data from all versions of the on-premises environment.
+
+
+## Proposal
+
+### Data Sending Flow
+
+All of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Data will be sent when one of the following happens:
+1. A cluster installation attempt finishes (regardless of success or failure)
+2. A shutdown trigger: cluster is getting deleted, all hosts are removed from a cluster, or the API server is getting shut down
+3. An interval of time has passed and the cluster has not changed state (meaning the installation is abandoned/forgotten), just send the data we have until that point
+
+Before the data is sent, Assisted Service will get a remote configuration from an endpoint we specify.
+
+Sending data should not interfere with customer workloads and deployment
+
+### Assisted-Service Changes 
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query and send data (data sending flow event 1 above)
+- Needs code changes when a cluster is deleted, all hosts are removed from a cluster, and the API server is shutting down to query and send data (data sending flow event 2 above)
```

Comment:
i would not bother with this at this point. same about the next bullet. we can just decide that success / fail / pending user action are finite states.